core/src/main/java/org/apache/calcite/rex/RexInterpreter.java

```diff
@@ -175,6 +175,14 @@ public Comparable visitCall(RexCall call) {
       return ceil(call, values);
     case EXTRACT:
       return extract(call, values);
+    case IS_DISTINCT_FROM:
```

Comment:
Please merge that `case` branches with relevant `EQUALS` and `NOT_EQUALS` via `// fall through`